pkglistgen/tool.py

```diff
@@ -753,27 +756,35 @@ def update_and_solve_target(
         if engine == Engine.product_composer:
             self.output_dir = productcompose_dir
         elif engine == Engine.legacy:
-            self.output_dir = product_dir
+            if not only_update_weakremovers:
```

Comment:
Shouldn't this be independent of the engine?

Reply:
I limits `--only-update-weakremovers` to legacy engine only, see changes in cli.py

Reply:
see https://github.com/openSUSE/openSUSE-release-tools/blob/master/pkglistgen/cli.py#L79 , if product_composer engine in pkglistgen would not handle 000release-packages, why should `--only-update-weakremovers` to be independednt of the engine?

Reply:
I'm not actually sure why, because even with product composer there is 000release-packages: https://src.suse.de/products/SLES/src/branch/16.0/000release-packages (note: internal link)

@adrianschroeter maybe you remember?

Reply:
Have a existing 000release-packages when pkglistgen with product composer is on that is fine, 000release-packages just a package name and it builds `-release` binary package, the point is who and how does the content of 000release-packages be generated, per 000release-pcakges in SLFO:Main:Staging:A it was not done by product composer, as well as not https://src.suse.de/products/SLES/src/branch/16.0/000release-packages, maybe I'm wrong but changelog says that. And you can not find `XXX-release.spec.in` in 000pacakge-groups in those projects.

Reply:
Right, so in that case it might just be managed manually, including weakremovers? Hm.

Reply:
That's true for Leap 16 atm, to manage release package manually, that's why I needed a better way here to handle weakremovers...

Reply:
Do you know what SLES is doing?

Reply:
> Do you know what SLES is doing?

I don't know will the thing gonna be changed in SLES16, but in SLE15, weakremovers.inc is not rely on 000update-repos, they have a manual maintained weakremovers.inc in 000packages-group, and copy it to 000release-pacakges by pkglistgen. See https://github.com/openSUSE/openSUSE-release-tools/blob/master/pkglistgen/tool.py#L863

Reply:
yes, indeed in SLES16/SL-Micro currently 000release-packages are managed manually

Reply:
Ok, so the productcompose + weakremovers.inc case is just not relevant yet.